src/gui/dialogs/select_orb_colors.cpp

```diff
@@ -0,0 +1,160 @@
+#include "gui/widgets/window.hpp"
+
+#include "preferences.hpp"
+#include "game_config.hpp"
+
+#include <boost/bind.hpp>
+
+namespace gui2 {
+REGISTER_DIALOG(select_orb_colors);
+
+tselect_orb_colors::tselect_orb_colors()
+	: show_unmoved_(preferences::show_unmoved_orb())
+	, show_partial_(preferences::show_partial_orb())
+	, show_moved_(preferences::show_moved_orb())
+	, show_ally_(preferences::show_allied_orb())
+	, show_enemy_(preferences::show_enemy_orb())
+	, unmoved_(preferences::unmoved_color())
+	, partial_(preferences::partial_color())
+	, moved_(preferences::moved_color())
+	, ally_(preferences::allied_color())
+	, enemy_(preferences::enemy_color())
+{
+}
+
+void tselect_orb_colors::pre_show(CVideo&, twindow& window)
+{
+	setup_orb_group("unmoved", show_unmoved_, unmoved_, window);
+	setup_orb_group("partial", show_partial_, partial_, window);
+	setup_orb_group("moved", show_moved_, moved_, window);
+	setup_orb_group("ally", show_ally_, ally_, window);
+	setup_orb_group("enemy", show_enemy_, enemy_, window);
+	
+	tbutton& reset = find_widget<tbutton>(&window, "orb_defaults", false);
+	event::connect_signal_mouse_left_click(reset, boost::bind(
+		&tselect_orb_colors::handle_reset_click,
+		this, boost::ref(window)
+	));
+}
+
+void tselect_orb_colors::display(CVideo& video)
+{
+	tselect_orb_colors dialog;
+	if(dialog.show(video)) {
+		preferences::set_show_unmoved_orb(dialog.show_unmoved_);
+		preferences::set_show_partial_orb(dialog.show_partial_);
+		preferences::set_show_moved_orb(dialog.show_moved_);
+		preferences::set_show_allied_orb(dialog.show_ally_);
+		preferences::set_show_enemy_orb(dialog.show_enemy_);
+
+		preferences::set_unmoved_color(dialog.unmoved_);
+		preferences::set_partial_color(dialog.partial_);
+		preferences::set_moved_color(dialog.moved_);
+		preferences::set_allied_color(dialog.ally_);
+		preferences::set_enemy_color(dialog.enemy_);
+	}
+}
+
+void tselect_orb_colors::setup_orb_group(const std::string& base_id, bool& shown, std::string& color, twindow& window, bool connect)
+{
+	ttoggle_button& toggle = find_widget<ttoggle_button>(&window, "orb_" + base_id + "_show", false);
+	toggle.set_value_bool(shown);
+	if(connect) {
+		event::connect_signal_mouse_left_click(toggle, boost::bind(
+			&tselect_orb_colors::handle_toggle_click,
+			this,
+			boost::ref(shown)
+		));
+	}
+
+	tgrid& selection = find_widget<tgrid>(&window, "orb_" + base_id + "_selection", false);
+	std::vector<ttoggle_button*>& group = groups_[base_id];
+
+	using iterator::twalker_;
+	twalker_* iter = selection.create_walker();
+	while(!iter->at_end(twalker_::child)) {
+		twidget* next = iter->get(twalker_::child);
+		if(ttoggle_button* button = dynamic_cast<ttoggle_button*>(next)) {
+			group.push_back(button);
+			if(button->id().rfind("_" + color) != std::string::npos) {
+				button->set_value_bool(true);
+			} else {
+				button->set_value_bool(false);
+			}
+			if(connect) {
+				event::connect_signal_mouse_left_click(*button, boost::bind(
```

Comment:
It's not hurting anyone. I think there was a reason or something for it to be there.